android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -87,18 +90,19 @@ public WebRTCModule(ReactApplicationContext reactContext) {
             }
         }
 
-        if (adm == null) {
-            adm = JavaAudioDeviceModule.builder(reactContext).setEnableVolumeLogger(false).createAudioDeviceModule();
-        }
-
         Log.d(TAG, "Using video encoder factory: " + encoderFactory.getClass().getCanonicalName());
         Log.d(TAG, "Using video decoder factory: " + decoderFactory.getClass().getCanonicalName());
 
-        mFactory = PeerConnectionFactory.builder()
-                           .setAudioDeviceModule(adm)
-                           .setVideoEncoderFactory(encoderFactory)
-                           .setVideoDecoderFactory(decoderFactory)
-                           .createPeerConnectionFactory();
+        if (adm != null) {
+            // Only eagerly initialize the PeerConnectionFactory 
```

Comment:
This logic preserves dependency injection. If some caller specifies an explicit `adm` to use, then we should use it here. Otherwise we'd have to make `adm` a class member and cache it until `getPeerConnectionFactory()` is called.